packages/protocol/contracts/stability/PairManager.sol

```diff
@@ -0,0 +1,290 @@
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import { IPairManager } from "./interfaces/IPairManager.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+import { IMentoExchange } from "./interfaces/IMentoExchange.sol";
+
+import { StableToken } from "./StableToken.sol";
+
+import { UsingRegistry } from "../common/UsingRegistry.sol";
+import { Initializable } from "../common/Initializable.sol";
+import { FixidityLib } from "../common/FixidityLib.sol";
+
+// TODO: Remove when migrating to mento-core. Newer versions of OZ-contracts have this interface
+interface IERC20Metadata {
+  function symbol() external view returns (string memory);
+}
+
+/**
+ * @title PairManager
+ * @notice The pair manager allows the creation of virtual pairs that are tradeable via Mento.
+ */
+contract PairManager is IPairManager, Initializable, UsingRegistry {
+  using FixidityLib for FixidityLib.Fraction;
+
+  /* ==================== State Variables ==================== */
+
+  // Address of the broker contract.
+  address public broker;
+
+  // Maps a pair id to the corresponding pair struct.
+  // pairId is in the format "stableSymbol:collateralSymbol:exchangeName"
+  mapping(bytes32 => Pair) public pairs;
+  // Tracks whether or not a pair with the given pairId exists.
+  mapping(bytes32 => bool) public isPair;
+
+  // TODO: Implement getPairs that takes in asset addresses & not exchange
+
+  // Tracks the total fraction of reserve collateral
+  // that has been allocated to existing virtual pairs.
+  mapping(address => FixidityLib.Fraction) public allocatedCollateralFractions;
+
+  // TODO: https://github.com/mento-protocol/mento-general/issues/60
+  // Tracks the total fraction of reserve stable that has been allocated to existing virtual pairs.
+  mapping(address => FixidityLib.Fraction) public allocatedStableFractions;
+
+  /* ==================== Constructor ==================== */
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialization.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice Allows the contract to be upgradable via the proxy.
+   * @param _broker The address of the broker contract.
+   * @param registryAddress The address of the Celo registry.
+   */
+  function initilize(address _broker, address registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
```

Comment:
In the Broker I went with just having a `reserve` variable in storage and moved away from the registry altogether. Wdyt?

Reply:
I'd definitely prefer that. Will update and do the same here.